features/256-auth-gear-interaction-overview-in-next/api.md

```diff
@@ -172,25 +176,38 @@ Followings are hooks of auth actions:
-| `roles` | `before_change_roles_sync(user, context)`<br/>`before_change_roles(user, context)`<br/>`after_change_roles_sync(user, context)`<br/>`after_change_roles(user, context)` |
 | `logout` | `before_logout_sync(user, context)`<br/>`before_logout(user, context)`<br/>`after_logout_sync(user, context)`<br/>`after_logout(user, context)` |
-| `change_password` | `before_change_password_sync(user, context)`<br/>`before_change_password(user, context)`<br/>`after_change_password_sync(user, context)`<br/>`after_change_password(user, context)` |
-| `reset_password` | `before_reset_password_sync(user, context)`<br/>`before_reset_password(user, context)`<br/>`after_reset_password_sync(user, context)`<br/>`after_reset_password(user, context)` |
-| `verify` | `before_verify_sync(user, context)`<br/>`before_verify(user, context)`<br/>`after_verify_sync(user, context)`<br/>`after_verify(user, context)` |
-| `unverify` | `before_unverify_sync(user, context)`<br/>`before_unverify(user, context)`<br/>`after_unverify_sync(user, context)`<br/>`after_unverify(user, context)` |
-| `update_metadata` | `before_update_metadata_sync(user, original_user, context)`<br/>`before_update_metadata(user, original_user, context)`<br/>`after_update_metadata_sync(user, original_user, context)`<br/>`after_update_metadata(user, original_user, context)` | 
+| `roles` | `before_roles_changed_sync(user, original_user, context)`<br/>`before_roles_changed(user, original_user, context)`<br/>`after_roles_changed_sync(user, original_user, context)`<br/>`after_roles_changed(user, original_user, context)` |
+| `enable` | `before_enable_changed_sync(user, original_user, context)`<br/>`before_enable_changed(user, original_user, context)`<br/>`after_enable_changed_sync(user, original_user, context)`<br/>`after_enable_changed(user, original_user, context)` |
+| `password` | `before_password_changed_sync(user, original_user, context)`<br/>`before_password_changed(user, original_user, context)`<br/>`after_password_changed_sync(user, original_user, context)`<br/>`after_password_changed(user, original_user, context)` |
+| `verify` | `before_verify_changed_sync(user, original_user, context)`<br/>`before_verify_changed(user, original_user, context)`<br/>`after_verify_changed_sync(user, original_user, context)`<br/>`after_verify_changed(user, original_user, context)` |
+| `metadata` | `before_metadata_changed_sync(user, original_user, context)`<br/>`before_metadata_changed(user, original_user, context)`<br/>`after_metadata_changed_sync(user, original_user, context)`<br/>`after_metadata_changed(user, original_user, context)` |
+| `user` | `before_user_changed_sync(user, original_user, context)`<br/>`before_user_changed(user, original_user, context)`<br/>`after_user_changed_sync(user, original_user, context)`<br/>`after_user_changed(user, original_user, context)` |
```

Comment:
`user_updated`? Feel a bit strange, "user changed or user updated" sounds like it is the users have changed to something else, while I know what it meant is "user object updated". We need a better name. Some idea:

- `[before|after]_user_object_updated`
- `user_profile_sync` / `user_profile_async` (this one is not consistent and don't have a "before/after" prefix, the idea is this hook is special and only for syncing profile to DB, which is always "after")